plugins/modules/zos_find.py

```diff
@@ -0,0 +1,791 @@
+
+    Returns:
+        set[str] -- The remaining data sets that have not been excluded
+    """
+    for ds in set(data_set_list):
+        for ex_pat in excludes:
+            if _match_regex(module, ex_pat, ds):
+                data_set_list.remove(ds)
+                break
+    return data_set_list
+
+
+def _age_filter(ds_date, now, age):
+    """Determine whether a given date is older than 'age'
+
+    Arguments:
+        ds_date {str} -- The input date in the format YYYY/MM/DD
+        now {float} -- The time elapsed since the last epoch
+        age {int} -- The age, in days, to compare against
+
+    Returns:
+        bool -- Whether 'ds_date' is older than 'age'
+    """
+    year, month, day = map(int, ds_date.split("/"))
+    if year == "0000":
+        return age >= 0
+
+    # Seconds per day = 86400
+    ds_age = datetime.datetime(year, month, day).timestamp()
+    if age >= 0 and (now - ds_age) / 86400 >= abs(age):
+        return True
+    elif age < 0 and (now - ds_age) / 86400 <= abs(age):
+        return True
+    return False
+
+
+def _get_creation_date(module, ds):
+    """Retrieve the creation date for a given data set
+
+    Arguments:
+        module {AnsibleModule} -- The Ansible module object being used
+        ds {str} -- The name of the data set
+
+    Returns:
+        str -- The data set creation date in the format "YYYY/MM/DD"
+    """
+    rc, out, err = mvs_cmd.idcams(
+        "  LISTCAT ENT('{0}') HISTORY".format(ds), authorized=True
+    )
+    if rc != 0:
+        module.fail_json(
+            msg="Non-zero return code received while retrieving data set age",
+            rc=rc, stderr=err, stdout=out
+        )
+    out = re.findall(r"CREATION-*[A-Z|0-9]*", out.strip())
+    years, days = "".join(re.findall(r"-[A-Z|0-9]*", out)).replace("-", "").split(".")
+    days = int(days)
+    days_per_month = 30.4167
+    return "{0}/{1}/{3}".format(
+        years,
+        math.ceil(days / days_per_month),
+        math.ceil(days % days_per_month)
+    )
+
+
+def _size_filter(ds_size, size):
+    """ Determine whether a given size is greater than the input size
+
+    Arguments:
+        ds_size {int} -- The input size, in bytes
+        size {int} -- The size, in bytes, to compare against
+
+    Returns:
+        bool -- Whether 'ds_size' is greater than 'age'
+    """
+    if size >= 0 and ds_size >= abs(size):
+        return True
+    if size < 0 and ds_size <= abs(size):
+        return True
+    return False
+
+
+def _match_regex(module, pattern, string):
+    """ Determine whether the input regex pattern matches the string
+
+    Arguments:
+        module {AnsibleModule} -- The Ansible module object being used
+        pattern {str} -- The regular expression to match
+        string {str} -- The string to match
+
+    Returns:
+        re.Match -- A Match object that matches the pattern to string
+    """
+    try:
+        return re.fullmatch(pattern, string, re.IGNORECASE)
```

Comment:
We might get errors on this during 2.7 linting, as fullmatch was not available

Reply:
In some other code that uses fullmatch, I specify if we are running python 2 to `from re import match as fullmatch` so we can continue to take advantage of fullmatch (since z/OS target will always be running 3.X anyways)